src/test/poll/test_poll.rs

```diff
@@ -197,6 +200,22 @@ fn test_poll_args_common(
 
         let instant_before = std::time::Instant::now();
 
+        unsafe {
+            signal::sigaction(
+                Signal::SIGALRM,
+                &signal::SigAction::new(
+                    test_utils::nop_sig_handler(),
+                    signal::SaFlags::empty(),
+                    signal::SigSet::empty(),
+                ),
+            )
+        }
+        .unwrap();
+        let mut interruptor = test_utils::Interruptor::new(signal_time, signal::Signal::SIGALRM);
+        if signal_time.is_zero() {
+            interruptor.cancel();
+        }
```

Comment:
Why cancel if it's zero? Couldn't the thread have already started and the signal already sent before this point?

Reply:
Possibly! Or it might not have and we'll interrupt the poll call without intending to. Cancelling it here would ensure the poll call won't be interrupted when signal_time is 0.

Changed this to use an `Option` and never create the Interruptor in the first place if it's unwanted, to be a bit clearer.